parachain/pallets/bridge/src/channel/inbound.rs

```diff
@@ -0,0 +1,117 @@
+use frame_support::{dispatch::DispatchResult, storage::StorageMap};
+use sp_std::{cell::Cell, marker::PhantomData, boxed::Box};
+use artemis_core::{ChannelId, AppId, Message, Verifier};
+use crate::{
+	Module,
+	Config,
+	InboundChannels,
+	primitives::{InboundChannel, InboundChannelData}
+};
+
+/// Construct an inbound channel object
+pub fn make_inbound_channel<T>(channel_id: ChannelId) -> Box<dyn InboundChannel<T::AccountId>>
+where
+	T: Config
+{
+	match channel_id {
+		ChannelId::Basic => Box::new(BasicInboundChannel::<T>::new()),
+		ChannelId::Incentivized => Box::new(IncentivizedInboundChannel::<T>::new()),
+	}
+}
+
+/// Basic Channel
+struct BasicInboundChannel<T: Config> {
+	#[allow(dead_code)]
+	channel_id: ChannelId,
+	#[allow(dead_code)]
+	storage: Storage<T>
+}
+
+impl<T: Config> BasicInboundChannel<T> {
+	fn new() -> Self {
+		Self {
+			channel_id: ChannelId::Basic,
+			storage: Storage::new(ChannelId::Basic)
+		}
+	}
+}
+
+impl<T: Config> InboundChannel<T::AccountId> for BasicInboundChannel<T> {
+	// This implementation is a WIP!
+	fn submit(&mut self, relayer: &T::AccountId, app_id: AppId, message: &Message) -> DispatchResult {
+		T::Verifier::verify(relayer.clone(), app_id, &message)?;
+		Module::<T>::dispatch(app_id.into(), message)
```

Comment:
Nonce check for replay protection should be here right?

Reply:
Similar for incentivized channel

Reply:
yeah, but wasn't planning to do it in this PR though. The focus for this PR is the channel API.

Reply:
Also the nonce check can't be implemented until the other side includes a nonce.

Reply:
The pr for the other side has this already but sure, merging and doing that in future PR sounds fine